tools/mo/openvino/tools/mo/utils/version.py

```diff
@@ -81,3 +85,64 @@ def extract_hash_from_version(full_version: str):
         return res[0]
     else:
         return None
+
+
+class SingletonMetaClass(type):
+    def __init__(self, cls_name, super_classes, dic):
+        self.__single_instance = None
+        super().__init__(cls_name, super_classes, dic)
+
+    def __call__(cls, *args, **kwargs):
+        if cls.__single_instance is None:
+            cls.__single_instance = super(SingletonMetaClass, cls).__call__(*args, **kwargs)
+        return cls.__single_instance
+
+
```

Comment:
It looks that we can refactor this file. I mean to make it more clear `simplify_version`, `get_simplified_mo_version`. I believe not all of functions above are needed and we can refactor and incorporate them inside the existing functions of `VersionChecker`. You can consider such refactoring in the next PR. it is up-to-you